client_server_test.go

```diff
@@ -1026,3 +1026,64 @@ func testClientCloseFinish(t *testing.T, s *cstServer) {
 		t.Fatal("timeout waiting for server to get client close")
 	}
 }
+
+func TestDialMaxHeaderSize(t *testing.T) {
+	s := newServer(t)
+	defer s.Close()
+	longHeader := http.Header{}
+	longHeader.Add("x-too-long-header", "toolong")
+
+	_, _, err := cstDialer.Dial(s.URL, longHeader)
+	if err != nil {
+		t.Fatalf("Expected err == nil, got %v", err)
+	}
+
+	// Make a local copy and set short max header size
+	cstDialer := cstDialer
```

Comment:
I don't know that there's real danger here, and I notice this pattern is used throughout this testing file, but I see that cstDialer contains a slice,

```
var cstDialer = Dialer{
	Subprotocols:     []string{"p1", "p2"},
	ReadBufferSize:   1024,
	WriteBufferSize:  1024,
	HandshakeTimeout: 30 * time.Second,
}
```

Which is not going to be copied by value into your local copy, if I have my go-knowledge straight. For that you'd want to use `copy`

However, in this situation, the Subprotocols isn't in play so I don't think you're really in any danger of being affected by other tests.

Reply:
Yeah, indeed it would be an issue if I used subprotocols but I just repeated the pattern used by the original library maintainers.